### DISCUSSION TITLE
About contributing to the golang version of openrarity

### DISCUSSION THREAD

**Discussion by kenshinvv** [OpenRarity/open-rarity#110] [Show and tell]
OpenRarity is an outstanding effort that is trying to smooth out the variance in rarity algorithms across platforms.

Although the official Python implementation is provided, as a basic library, it would be better if there were more common language implementations (such as Javascript, Golang, and Rust). Because the calculation of rarity is not limited to an offline script, it is likely to be used for real-time calculation by back-end services.

Therefore, I spent some time to provide the implementation of the Golang version: https://github.com/Base-Labs/openrarity-go.

I tried to make sure during the implementation that the design of functions, classes, and single tests are consistent with `OpenRarity/open-rarity`, which will make adding new features or locating differences very easy in the future. After a lot of work, it computes in perfect agreement with the official implementation.

Now I would like to contribute `openrarity-go` to OpenRarity for the following reasons.
1. OpenRarity is a group, and `open-rarity` is just a python implementation of it, which logically should contain implementations in more languages.
2. putting `openrarity-go` under `https://github.com/OpenRarity/` makes it easier for developers to discover and use it.
3. post maintenance of `openrarity-go` can be left to the community, but for now it looks like `v1.0.0` is more than enough to use.

Any response is worth waiting for.

**Comment by impreso** [comment]
Wow thanks for the effort will take a look at it!
